module/zfs/vdev_raidz.c

```diff
@@ -1440,6 +1440,38 @@ vdev_raidz_child_done(zio_t *zio)
 	rc->rc_skipped = 0;
 }
 
+boolean_t
+vdev_raidz_need_resilver(vdev_t *vd, uint64_t offset, size_t psize)
+{
+	uint64_t unit_shift = vd->vdev_top->vdev_ashift;
+	uint64_t dcols = vd->vdev_children;
+	uint64_t nparity = vd->vdev_nparity;
+	uint64_t b = offset >> unit_shift;
+	uint64_t s = ((psize - 1) >> unit_shift) + 1;
+	/* The first column for this stripe. */
+	uint64_t f = b % dcols;
+	uint64_t c, devidx;
```

Comment:
Too bad the `vdev_raidz_map_alloc()` is allocation heavy, so this raidz geometry have to be calculated separately...